server_src/services/database/server-settings.service.js

```diff
@@ -0,0 +1,294 @@
+   * @returns {Object} Object containing the filtered settings / full settings
+   */
+  returnDatabaseValues({ filter = {}, options = {} } = {}) {
+    return ServerSettingsDB.find(filter, {}, options);
+  }
+  /**
+   * Returns the entire settings object from catch
+   * @returns {Object} Object from cache for all of OctoFarms settings
+   */
+  get entireSettingsObject() {
+    return this.currentServerSettings;
+  }
+
+  /**
+   * Returns the polling settings for OctoPrints websocket connection
+   * @returns {Object} Object from cache for OctoPrints websocket connection
+   */
+  get octoPrintWebsocketPollingSettings() {
+    return this.currentServerSettings["onlinePolling"];
+  }
+  /**
+   * Updates the polling settings for OctoPrints websocket connection in the database and then the cache.
+   * @param {Object} value containing the seconds key for OctoPrints websocket polling. String value of polling time.
+   * @throws {Error} value not provided or empty.
+   * @returns {Object} containing the new settings key
+   */
+  async updateOctoPrintWebsocketPollingSettings(value = {}) {
+    const model = new ServerSettingsDB(value);
+
+    const validatedModel = model.validateSync();
+    if (!!validatedModel) throw validatedModel;
+
+    //Filter out the current settings from cache
+    const filter = this.currentServerSettings._id;
+
+    // Find the filter and update the server settings with the new value...
+    const updatedSettings = await ServerSettingsDB.findOneAndUpdate(
+      { _id: filter },
+      { onlinePolling: value },
+      {
+        new: true
+      }
+    );
+    // Make sure to replicate the changes in cache.
+    this.currentServerSettings["onlinePolling"] = value;
+
+    return updatedSettings["onlinePolling"];
+  }
+  /**
+   * Returns the server specific settings object for OctoFarm
+   * @returns {Object} Object from cache for OctoFarms server settings
+   */
+  get systemSettings() {
+    return this.currentServerSettings["server"];
+  }
+  /**
+   * Updates the polling settings for OctoPrints websocket connection in the database and then the cache.
+   * @param {Object} value containing the seconds key for OctoPrints websocket polling. String value of polling time.
+   * @throws {Error} value not provided or empty.
+   * @returns {Object} containing the new settings key
+   */
+  async updateSystemSettings(value = {}) {
+    if (!value || isEmpty(value))
+      throw new Error("No value provided to update settings");
+
+    //Filter out the current settings from cache
+    const filter = this.currentServerSettings._id;
+
+    // Find the filter and update the server settings with the new value...
+    const updatedSettings = await ServerSettingsDB.findOneAndUpdate(
+      { _id: filter },
+      { server: value },
+      {
+        new: true
+      }
+    );
+    // Make sure to replicate the changes in cache.
+    this.currentServerSettings["server"] = value;
```

Comment:
A string prevents refactored renaming in an IDE like WebStorm. In general I'm not fond of string based code, although in Python we see it a lot sadly. What is your reasoning to do so here and beyond?